x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedTimingStats.java

```diff
@@ -85,28 +93,58 @@ public long getSearchCount() {
         return searchCount;
     }
 
+    public long getBucketCount() {
+        return bucketCount;
+    }
+
     public double getTotalSearchTimeMs() {
         return totalSearchTimeMs;
     }
 
+    public Double getAvgSearchTimePerBucketMs() {
+        return bucketCount > 0
+            ? totalSearchTimeMs / bucketCount
+            : null;
+    }
+
     public void incrementTotalSearchTimeMs(double searchTimeMs) {
         this.searchCount++;
         this.totalSearchTimeMs += searchTimeMs;
     }
 
+    public void setBucketCount(long bucketCount) {
+        this.bucketCount = bucketCount;
+    }
+
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(jobId);
         out.writeLong(searchCount);
+        out.writeLong(bucketCount);
         out.writeDouble(totalSearchTimeMs);
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        return toXContent(builder, false);
+    }
+
+    public XContentBuilder toXContentWithCalculatedFields(XContentBuilder builder) throws IOException {
+        return toXContent(builder, true);
+    }
+
+    private XContentBuilder toXContent(XContentBuilder builder, boolean includeCalculatedFields) throws IOException {
         builder.startObject();
         builder.field(JOB_ID.getPreferredName(), jobId);
         builder.field(SEARCH_COUNT.getPreferredName(), searchCount);
+        builder.field(BUCKET_COUNT.getPreferredName(), bucketCount);
         builder.field(TOTAL_SEARCH_TIME_MS.getPreferredName(), totalSearchTimeMs);
+        if (includeCalculatedFields) {
```

Comment:
You should take advantage of `ToXContent.Params params` and look for the appropriate option. That parameter exists exactly for situations like this. 

Consequently, `toXContentWithCalculatedFields` should probably go away.

Reply:
Done.

Thanks for this hint!